module4/projects/ApiCheckpoint.md

```diff
@@ -0,0 +1,119 @@
+</section>
+
+<section class='call-to-action' markdown='1'>
+
+**2. CRUD Endpoints**
+
+Implement API endpoints for the following CRUD operations:
+* Create a new product associated with a specific merchant
+* Retrieve a specific product by its primary key
+* Update an existing product
+* Delete a product by its primary key
+
+</section>
+
+<section class='call-to-action' markdown='1'>
+
+**3. Statistics Endpoints**
+
+Implement API endpoints for the following statistics:
+* Count of Products by Category (regardless of Merchant)
+* Average Price of Products for each merchant
+* Newest Product Release Date for each Merchant
+
+</section>
+
+<section class='instructor-note' markdown='1'>
+
+The next task will be difficult for most students to complete.  We want them to have the experience of going in to an interview with not a _whole_ challenge done.  This way we can see how they handle that situation.  Do they reach out? Do they clam up? We can then coach them on this scenario.
+
+</section>
+
+<section class='call-to-action' markdown='1'>
+
+**4. Error Handling**
+
+Properly handle and return appropriate HTTP status codes and error messages for various scenarios, such as invalid inputs or non-existing resources.
+
+</section>
+
+### Deadline and Submission
+
+<section class='instructor-note' markdown='1'>
+
+Depending on the calendar, the deadline may vary, but it should be before student's scheduled interviews!
+
+</section>
+
+You instructors will post a deadline for submissions in slack.
+
+Submit your challenge by:
+* Creating a PR from your main branch to the turingschool-examples main branch.
+* Include your Name in your PR title.
+* Include an overview of what you were able accomplish in the PR description.
+
+### Interview Time!
+
+The final step of a take-home challenge is an interview!  You will have an interview scheduled with an instructor and during that interview you will be asked questions related to APIs, Coding, and your work on the Takehome Challenge.  You *will* be asked the following questions, and you *may* be asked additional questions.
+
+1. What is a REST API, and how does it differ from other types of APIs?
+2. How did you handle different HTTP methods (GET, POST, PUT, DELETE) in your C#/.NET API, and why is this important for RESTful design?
+5. Explain the role of HTTP request bodies in REST APIs, and mention any specific C#/.NET libraries or methods you used to work with request bodies in your project.
+
+<section class='instructor-note' markdown='1'>
+
+<!-- Day 2 - Interview -->
+
+We would not all these questions with students, but my thought is that as part of this exercise, we can do a mock interview. Maybe we can make the rubric a bit more flexible - like would they have made it to the next round of the interview.  I think we can internally set a metric of completeness and their ability to answer or gracefully exit the question portion.
+
+
+1. What is a REST API, and how does it differ from other types of APIs?
```

Comment:
If you have any ideas for a question or two that we could ask that is something they should be able to answer that would be helpful! Or we could maybe only give two questions they will have time to prepare for?